database-commons/src/main/java/io/cdap/plugin/db/sink/ETLDBOutputFormat.java

```diff
@@ -191,6 +209,9 @@ private Connection getConnection(Configuration conf) {
     }
     return connection;
   }
+  public String constructUpsertQuery(String table, String[] fieldNames, String[] listKeys) {
```

Comment:
This method always returns 'null'?
what's the purpose here?

Reply:
This method is used to create Upsert query and the query for upsert is different for all plugins. This method is acting as a parent method and for postgreSQL plugins, this method is getting overridden for the upsert query creation for postgrSQL plugins. 

As per the requirement only PostgreSQL plugins were required to have upsert query. If in future, any other plugin is asked to have upsert query then the same procedure will be followed.

Reply:
this method should return this.constructUpsertQuery, so that child class method can be invoked, if this has child class reference

Reply:
done